pkg/controller/core.oam.dev/v1alpha2/application/revision.go

```diff
@@ -105,6 +89,21 @@ func (h *AppHandler) createResourcesConfigMap(ctx context.Context,
 	return h.r.Client.Create(ctx, cm)
 }
 
+// SprintComponentManifest formats and returns the resulting string.
+func SprintComponentManifest(cm *types.ComponentManifest) string {
+	cl := map[string]interface{}{
+		"StandardWorkload": string(util.MustJSONMarshal(cm.StandardWorkload)),
+	}
+
+	trs := []string{}
+	for _, tr := range cm.Traits {
+		trs = append(trs, string(util.MustJSONMarshal(tr)))
```

Comment:
we should write tests to cover these kinds of logic.

Reply:
This has been covered